app/backend/src/couchers/servicers/api.py

```diff
@@ -543,6 +540,27 @@ def ListFriends(self, request, context, session):
             user_ids=[rel.from_user.id if rel.from_user.id != context.user_id else rel.to_user.id for rel in rels],
         )
 
+    def RemoveFriend(self, request, context, session):
+        rel = session.execute(
+            select(FriendRelationship)
+            .where_users_column_visible(context, FriendRelationship.from_user_id)
+            .where_users_column_visible(context, FriendRelationship.to_user_id)
+            .where(
+                or_(
+                    FriendRelationship.from_user_id == request.user_id,
+                    FriendRelationship.to_user_id == request.user_id,
+                )
+            )
+            .where(FriendRelationship.status == FriendStatus.accepted)
+        ).scalar_one_or_none()
+
+        if not rel:
+            context.abort(grpc.StatusCode.FAILED_PRECONDITION, errors.NOT_FRIENDS)
+
+        session.delete(rel)
+
+        return empty_pb2.Empty()
+
```

Comment:
Does the FriendRelationship table store friends in both directions?

If so we need to make sure both directions are removed.

Since it says `scalar_one_or_none` seems it's only going to do one.

Reply:
It's only one way, from the `from` is the requester, `to` is the requestee.